framework/src/materials/Material.C

```diff
@@ -171,8 +175,26 @@ Material::getOutputs()
 void
 Material::computeProperties()
 {
-  for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
+  // If this Material has the _constant_on_elem flag set, we take the
```

Comment:
@YaqiWang I don't think we need the `preCompute()` in this case... because `computeQpProperties()` is only going to be called _once_... so you can just do your averaging in there.